server/config.go

```diff
@@ -64,21 +67,43 @@ type Config struct {
 	// MaxPeerCount for a region. default is 3.
 	MaxPeerCount uint64 `toml:"max-peer-count" json:"max-peer-count"`
 
-	BalanceCfg *BalanceConfig `toml:"balance" json:"balance"`
+	BalanceCfg BalanceConfig `toml:"balance" json:"balance"`
 
 	// Server advertise listening address for outer client communication.
 	// Host:Port
-	AdvertiseAddr string
+	AdvertiseAddr string `json:"-"`
 
 	// Only test can change it.
 	nextRetryDelay time.Duration
+
+	configFile string
 }
 
 // NewConfig creates a new config.
 func NewConfig() *Config {
-	return &Config{
-		BalanceCfg: newBalanceConfig(),
-	}
+	cfg := &Config{}
+	cfg.FlagSet = flag.NewFlagSet("pd", flag.ContinueOnError)
+	fs := cfg.FlagSet
+
+	fs.StringVar(&cfg.configFile, "config", "", "Config file")
+
+	fs.Uint64Var(&cfg.ClusterID, "cluster-id", 0, "initial cluster ID for the pd cluster")
+	fs.StringVar(&cfg.Name, "name", defaultName, "human-readable name for this pd member")
+	fs.StringVar(&cfg.DataDir, "data-dir", defaultDataDir, "path to the data directory (default 'default.{name}')")
+	fs.StringVar(&cfg.Host, "host", defaultHost, "host for outer traffic")
+	fs.Uint64Var(&cfg.Port, "port", defaultPort, "server port (deprecate later)")
+	fs.Uint64Var(&cfg.AdvertisePort, "advertise-port", 0, "advertise server port (deprecate later) (default '{port}')")
+	fs.Uint64Var(&cfg.HTTPPort, "http-port", defaultHTTPPort, "http port (deprecate later)")
+	fs.Uint64Var(&cfg.ClientPort, "client-port", defaultClientPort, "port for client traffic")
+	fs.Uint64Var(&cfg.AdvertiseClientPort, "advertise-client-port", 0, "advertise port for client traffic (default '{client-port}')")
+	fs.Uint64Var(&cfg.PeerPort, "peer-port", defaultPeerPort, "port for peer traffic")
+	fs.Uint64Var(&cfg.AdvertisePeerPort, "advertise-peer-port", 0, "advertise port for peer traffic (default '{peer-port}')")
+	fs.StringVar(&cfg.InitialCluster, "initial-cluser", defaultInitialCluster, "initial cluster configuration for bootstrapping (default '{name}=http://{host}:{advertise-peer-port}')")
```

Comment:
I think we can remove defaultInitialCluster  and leave empty here.
So if user doesn't set it, default used, but here default may be not defaultInitialCluster.